lnwallet/chanfunding/coin_select.go

```diff
@@ -214,3 +214,136 @@ func CoinSelectSubtractFees(feeRate chainfee.SatPerKWeight, amt,
 
 	return selectedUtxos, outputAmt, changeAmt, nil
 }
+
+// CoinSelectUpToAmount attempts to select coins such that we'll spend up
```

Comment:
Why can't `CraftSweepAllTx` be used for this purpose?

Reply:
This also doesn't lock outpoints at all, which can lead to race conditions.

Reply:
As far as I can see, the `CraftSweepAllTx` will spend _all_ inputs. But for opening channels there is a maximum constant `lnd.MaxFundingAmount` that a channel funding is not allowed to exceed. 
It would probably be possible to change `CraftSweepAllTx` to allow to specify a maximum amount, but would logically not make sense in a sweep package?

In addition, opening channels requires sending many messages. E.g. before attempting to open a channel LND will check if the remote node is responding, using dust limits depending on the chain used, etc. From the workflow of opening a channel it wasn't obvious to me how to get the funding output to use as "delivery" address, without specifying the funding amount in the first place. And I also didn't want to change this workflow when `fundall` is used.

Regarding locking of outpoints, as far as I see this should already happen in the `wallet_assembler.go` like for the other coin select subroutines:
https://github.com/lightningnetwork/lnd/blob/38b521d87d3fd9cff628e5dc09b764aeabaf011a/lnwallet/chanfunding/wallet_assembler.go#L299-L306